parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+                else:
+                    notices_others.append('mit %s' % notice['displayName'])
+
+    if sub_notices:
+        if len(notices_allergens) > 0:
+            notices_allergens[0] = 'enthält %s' % notices_allergens[0]
+        if len(notices_allergens) > 1:
+            notices_allergens[-2] = (
+                '%s und %s' % (notices_allergens[-2], notices_allergens[-1]))
+            del notices_allergens[-1]
+        if len(notices_others) > 0:
+            notices_others[0] = 'mit %s' % notices_others[0]
+        if len(notices_others) > 1:
+            notices_others[-2] = (
+                '%s und %s' % (notices_others[-2], notices_others[-1]))
+            del notices_others[-1]
+
+    return (notices_special + notices_allergens +
+            notices_others + notices_others_last)
+
+
+def build_notes(notices, components):
+    global base_data
+
+    components_all = []
+
+    notices_list = get_notices(notices)
+
+    for component in components:
+        component_string = 'dazu ' + component['name']
+        if len(component['notices']) > 0:
+            component_notices_list = get_notices(component['notices'], True)
+            component_string += ' (%s)' % ', '.join(component_notices_list)
+        components_all.append(component_string)
+
+    return components_all + notices_list
+
+
+def build_hours(opening_hours):
+    if not opening_hours:
+        return []
+
+    start = datetime.datetime.strptime(
+        opening_hours['start'], UTC_DATE_STRING).time()
+    end = datetime.datetime.strptime(
+        opening_hours['end'], UTC_DATE_STRING).time()
+    return [
+        'verfügbar von %s bis %s Uhr' %
+        (datetime.time.strftime(start, "%H:%M"),
+            datetime.time.strftime(end, "%H:%M"))
+    ]
+
+
+def build_location(description):
+    return ['Ort: %s' % description] if description else []
+
+
+def parse_url(url, today=False):
+    global base_data
+
+    canteen = LazyBuilder()
+    with urlopen(url) as response:
+        data = json.loads(response.read().decode())
+
+    for day in data['days']:
+        date = day.get('date')
+
+        if today and (datetime.datetime.now().date() !=
+                      datetime.datetime.strptime(date, UTC_DATE_STRING).date()):
+            continue
+
+        for counter in day['counters']:
+            counter_name = counter['displayName']
+            counter_description = counter['description']
+            counter_hours = counter.get('openingHours')
+            counter_color = counter.get('color')
+
+            for meal in counter['meals']:
+                if 'knownMealId' in meal:
+                    print('knownMealId: %s' % meal['knownMealId'])
+
+                meal_name = meal['name']
+                if 'category' in meal:
+                    meal_name = '%s: %s' % (meal['category'], meal_name)
+
+                meal_notes = (
+                    build_location(counter_description) +
```

Comment:
Typically yes, but the API doesn't promise that, this might also contain other human-readable information.

Reply:
Okay, I'm not quite sure how to handle this case of checking whether it is a location or not. The `build_location` function could check for all known locations prior to prepending "Ort: ", but this would be hardcoded information again.

So better drop the `build_location` logic or is there a way changing the API to always get a location with a predefined key?

Reply:
I think we can leave it for now, if this is typically the location and change it if there are actually other types of descriptions someday.

Reply:
I've added a little comment explaining this.